tvm/src/codegen/hlsc/port_direction.h

```diff
@@ -0,0 +1,107 @@
+/*!
+ *  Copyright (c) 2021 by Contributors
+ * \file port_direction.h
+ * \brief Implements an IR pass to infer SystemC module port directions
+ */
+#ifndef CODEGEN_HLSC_PORT_DIRECTION_H_
+#define CODEGEN_HLSC_PORT_DIRECTION_H_
+
+#include <tvm/ir.h>
+#include <tvm/ir_pass.h>
+#include <tvm/ir_visitor.h>
+
+namespace TVM {
+namespace ir {
+
+void canonicalize_string(std::string& s);
+enum class PortType { ChannelIn, ChannelOut, Memory, OffChipMemory, Default };
+
+class PortDirection : public IRVisitor {
```

Comment:
Put these two IRvisitors (`PortDirection` and `Hierarchy`) into `ir_utils.h`. Also it is better to rename them following the `er` convention that is used elsewhere, like `PortDirectionFinder`

Reply:
I just solved this issue in https://github.com/cornell-zhang/heterocl/pull/451/commits/9fd7a702a8556a9e1e25915191864ae523d0d975. Please review and merge if there's no further comments. Many thanks ;)